### PR TITLE
Add k-prefix Problem to CP 

### PR DESCRIPTION
**Pull request by Bharati2612**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Given a list of integers nums and an integer k, return the maximum possible i where nums[0] + nums[1] + ...  + nums[i] ≤ k. Return -1 if no valid i exists.

### Checklist

- [Y] I've read the contribution guidelines.
- [Y] I've checked the issue list before deciding what to submit.
- [Y] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue number #1905

### CODE REVIEW

**Comment by srimanikantapalakollu** [comment]
@Bharati2612 Your PR has a conflict with the base branch, please make the PR properly.

**Comment by Bharati2612** [comment]
> @Bharati2612 Your PR has a conflict with the base branch, please make the PR properly.

@srimani-programmer Done

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -47,6 +47,8 @@
 - [NQueens](cp/NQueens.cpp)
 - [Longest Subarray with Zero Sum](cp/ZeroSum.cpp)
 - [Next Largest Permutation](cp/NextPermutation.cpp)
+- [K-Prefix Algorithm](cp/K-Prefix.cpp)
+
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md:51]: I don't see the README update for `Partition_Problem.cpp`

File: C-Plus-Plus/cp/K_Prefix.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
+//K Prefix
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/K_Prefix.cpp:1]: Don't use Single-Line Comments and Multi-Line Comments inconsistently.

File: C-Plus-Plus/cp/K_Prefix.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+Input: 
+Enter the value of k: 4
+Enter the list of integers: 3 -5 4 1 6
+
+Output:
+Max Possible i is: 3
+
+Time Complexity: O(n)  Size of the array
+
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/K_Prefix.cpp:58]: Add a new line at the End of Code

File: C-Plus-Plus/cp/K_Prefix.cpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	{
+		nums.push_back(x);
+	}
+}
+int main()
+{
+	input();
+	int max_i=solve();
+	cout<<"Max Possible i is: "<<max_i;
+}
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/K_Prefix.cpp:45]: Format the Code aptly using Clang.

File: C-Plus-Plus/dp/Matrix_Chain_Multiplication.cpp [hunk 5]
```diff
@@ -98,4 +98,8 @@ Output
 
 Minimum number of multiplications are: 38000
 The way which give the minimum number of scalar multiplications is: (((A1A2)A3)A4)
+
+Complexity:
+Time:n^3
+Space:n^2
 */
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Matrix_Chain_Multiplication.cpp:105]: Add a new line at the end of the Code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by atarax665** [comment]
@Bharati2612 Are you still working on this PR?

**Comment by HarshCasper** [comment]
Closing the PR because of no updates.

**Pull request closed without merging**